OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -214,7 +214,7 @@ string IOrderVoice.VoicePhraseForOrder(Actor self, Order order)
 			return order.OrderString == attackOrderName || order.OrderString == forceAttackOrderName ? Info.Voice : null;
 		}
 
-		public abstract Activity GetAttackActivity(Actor self, Target newTarget, bool allowMove, bool forceAttack);
+		public abstract Activity GetAttackActivity(Actor self, Target newTarget, bool allowMove, bool forceAttack, bool ordered = false);
```

Comment:
Can you elaborate on the need for the `ordered` flag?
Plumbing this through to the activities so they can suppress their target lines makes up a significant fraction of the changes here, so i'd like to make sure there isn't a more concise solution.

It looks like this is mainly for `AttackMove`, but can't we solve that at the source by having that simply not chain through to the child attack activities?

Reply:
`AttackMoveActivity` already doesn't chain through to child activities (like most activities) so it isn't needed for that. It is to prevent `AutoTarget` from drawing target lines. 

The reasoning being: Opportunity attacks during a regular move don't draw target lines, automatic attacks during attack-move don't draw targeting lines, so it would be consistent if automatic attacks while stationary also don't draw target lines. Only when an attack is explicitly ordered does it draw a target line.

Reply:
Ok, that makes sense and I can't think of a better alternative at this point. Can we please split this into its own commit?

Reply:
We may want to show the target lines if `!ordered` but `allowMove` because it would look weird for the unit to start chasing the target around the map without having a target line.

Reply:
Can we please replace ordered with a targetLineColor? This would make the attack activities consistent with the others that allow the caller to override the line behaviour, and would let us give e.g. heal weapons a different line colour in a follow up pr.

Reply:
about the `allowMove` part:

I admit it would be awkward to not know what the unit is chasing there. On the other hand, this is the same situation as with `attackMove` so it would be consistent. It is also a bit awkward to see order lines appearing when you didn't order anything.

In the long term, as I indicated in the OP, I would like to reintroduce target lines as a completely separate thing from order lines. It would be a 'weaker' (thinner and/or dashed and/or transparent) line, called from the trait code and pointing to whatever is the current (opportunity) attack target. But this will have to wait for the `Attack*` refactor. 

I'm not sure what the best option for the short term is though.

Completely agree on the `targetLineColor` part.

Reply:
Replaced `ordered` with `targetLineColor` and made the attack colour yaml configurable.